joplin/users/tests/tests.py

```diff
@@ -133,3 +137,63 @@ def test_make_superuser_with_roles_and_department(department):
     form = make_user_form(user_data)
     is_valid = form.is_valid()
     assert is_valid is True
+
+
+@pytest.mark.django_db
+def test_editor_makes_page_under_department(editor, rf):
+    page_data = {
+        "publish_janis_branch_for_pr": "pytest",
+        "preview_janis_branch_for_pr": "pytest",
+        "slug": "pytest",
+        "title": "pytest",
+    }
+    HomePageFactory.create(**page_data)
+    # editor creates page
+    body = {
+            "type": "information",
+            "jwtToken": "",
+            "title": "Test Page",
+            "topic": "",
+            "department": "",
+    }
+    request = rf.post('admin/pages/new_from_modal', body, content_type='application/json')
+    request.user = editor
+
+    # assert page is created
+    response = new_page_from_modal(request)
+    assert response.status_code is 200
+    # assert page has a department that matches editor department
+    response_json = json.loads(response.content.decode('utf-8'))
+    page_pk = response_json['id']
+    created_page = InformationPage.objects.get(id=page_pk)
+    editor_department = editor.groups.get(name="Kitchen sink department")
+    assert created_page.departments()[0].department.pk is editor_department.department.pk
+
+
+@pytest.mark.django_db
+def test_admin_can_make_departmentless_page(superadmin, rf):
+    page_data = {
+        "publish_janis_branch_for_pr": "pytest",
+        "preview_janis_branch_for_pr": "pytest",
+        "slug": "pytest",
+        "title": "pytest",
+    }
+    HomePageFactory.create(**page_data)
```

Comment:
should I make this each time? Or should I set up the homepage only once before running the tests? (if so, where do I put it?)

Reply:
For now it’s just the two times so I’d say it’s fine

Reply:
We've got a fixture for a dummy HomePage in pages/home_page/fixtures/test_cases/pytest.py

Reply:
I saw the fixture but without me creating this page, I'd get an error saying there was no home page